lib/Dialect/DC/Transforms/DCPrintDot.cpp

```diff
@@ -0,0 +1,338 @@
+      .Case([&](dc::JoinOp) { n.nodeType = joinOp; })
+      .Case([&](dc::MergeOp) { n.nodeType = mergeOp; })
+      .Case([&](dc::PackOp) { n.nodeType = packOp; })
+      .Case([&](dc::SelectOp) { n.nodeType = selectOp; })
+      .Case([&](dc::SinkOp) { n.nodeType = sinkOp; })
+      .Case([&](dc::SourceOp) { n.nodeType = sourceOp; })
+      .Case([&](dc::ToESIOp) { n.nodeType = toESIOp; })
+      .Case([&](dc::UnpackOp) { n.nodeType = unpackOp; });
+
+  return n;
+}
+
+/// creates node in the dataflow graph for Comb operations
+DotNode createCombNode(
+    Operation &op,
+    llvm::SmallVector<std::pair<mlir::Value, std::string>> &valuesMap) {
+
+  DotNode n = {null, valueToName(op.getOperands(), valuesMap, false),
+               valueToName(op.getOperands(), valuesMap, false)};
+
+  TypeSwitch<mlir::Operation *>(&op)
+      .Case([&](comb::AddOp) { n.nodeType = addOp; })
+      .Case([&](comb::AndOp) { n.nodeType = andOp; })
+      .Case([&](comb::XorOp) { n.nodeType = xorOp; })
+      .Case([&](comb::OrOp) { n.nodeType = orOp; })
+      .Case([&](comb::MulOp) { n.nodeType = mulOp; })
+      .Case([&](comb::MuxOp) { n.nodeType = muxOp; })
+      .Case<comb::ICmpOp>([&](auto op) {
+        switch (op.getPredicate()) {
+        case circt::comb::ICmpPredicate::eq: {
+          n.nodeType = eqOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::ne: {
+          n.nodeType = neOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::sgt: {
+          n.nodeType = gtOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::sge: {
+          n.nodeType = geOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::slt: {
+          n.nodeType = ltOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::sle: {
+          n.nodeType = leOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::ugt: {
+          n.nodeType = gtOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::uge: {
+          n.nodeType = geOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::ult: {
+          n.nodeType = leOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::ule: {
+          n.nodeType = ltOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::ceq: {
+          n.nodeType = eqOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::cne: {
+          n.nodeType = neOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::weq: {
+          n.nodeType = eqOp;
+          break;
+        }
+        case circt::comb::ICmpPredicate::wne: {
+          n.nodeType = neOp;
+          break;
+        }
+        }
+      });
+  return n;
+}
+
+namespace {
+/// Emit the dot nodes
+struct DCDotPrintPass : public circt::dc::impl::DCDotPrintBase<DCDotPrintPass> {
+  DCDotPrintPass(llvm::raw_ostream &os) : os(os) {}
```

Comment:
```suggestion
  explicit DCDotPrintPass(llvm::raw_ostream &os) : os(os) {}
```
Nit: Without this, C++ allow to use this constructor for implicit type conversion. Fantastic language design, btw ;P